### ISSUE TITLE
loss formulations

### ISSUE THREAD

**Issue by lihuiliullh** [yuh-yang/KGCL-SIGIR22#15]
May I know which loss formulations in the paper these two images correspond to?

![image](https://user-images.githubusercontent.com/53407011/226070928-ad8dce2d-ab1e-4cd7-a71f-0c64a38c0d65.png)

![image](https://user-images.githubusercontent.com/53407011/226071047-f54aa2e5-a80b-4045-9f09-8419070b604a.png)

**Comment by yuh-yang** [comment]
Hi, @lihuiliullh !

`reg_loss` computes the L2 regularization for batch user and item embeddings.

`loss` corresponds to the BPR loss for recommendation task.

**Comment by lihuiliullh** [comment]
@yuh-yang 
May I know why you use L2 regularization for batch user and item embeddings? Does this trick can improve performance?

**Comment by lihuiliullh** [comment]
I also notice that for BPR loss, the formulation in the paper is 
![image](https://user-images.githubusercontent.com/53407011/226144404-c4533c4a-b781-48d1-8308-bf4beddf3059.png)

But in your code, it is softplus(-(pos_scores - neg_scores)).

Are these two the same?

**Comment by yuh-yang** [comment]
1. Generally L2 regularization is effective against overfitting. This batch-wise usage follows NGCF and LightGCN.

2. Using `softplus` instead of `logsigmoid` in BPR loss is a common practice to avoid NaN loss when, on some cases during training, the model fails to perform well by scoring negative samples very high.

Referring to this: https://github.com/xiangwang1223/neural_graph_collaborative_filtering/issues/17

**Comment by yuh-yang** [comment]
Closed for being inactive.
